geopandas/geodataframe.py

```diff
@@ -473,7 +478,22 @@ def to_json(self, na="null", show_bbox=False, **kwargs):
           feature individually so that features may have different properties.
         - ``keep``: output the missing entries as NaN.
         """
-        return json.dumps(self._to_geo(na=na, show_bbox=show_bbox), **kwargs)
+        if self.crs and not to_wgs84 and not self.crs == 4326:
+            warnings.warn(
+                "GeoJSON specification requires WGS84 CRS. Geometry will "
+                "be automatically re-projected in future versions of GeoPandas. "
+                "To keep exsting behaviour use 'to_wgs84=False'. "
```

Comment:
Maybe we should make the actual default now `None`, so we can detect if the user specified `True` or `False` and already do the future behaviour now.

Which also raises the question: what future behaviour do we want? Convert the crs silently, or still raise a warning when we do?

Reply:
I would do automatic re-projection with a warning that it has happened in the future.

I'll change the default to None so we can then just change it to True and remove warning.

Reply:
A slight alternative would be to keep the default as None in the future (but still change to automatic reprojection at that point), which has a default of reprojecting + warning, which gives the ability to users to do `to_wgs84=True` which has the same behaviour but silences the warning. 
I think it is nice to give users a way to have it not raise a warning.

(in which case it's only the warning for `to_wgs84=True` that can be removed, or commented out to use for the default of None in the future)

Reply:
>  I think it is nice to give users a way to have it not raise a warning.

Although of course `gdf.to_crs(epsg=4326).to_json(..)` is also an option to do that ..

Reply:
Keeping None as default is an elegant option. I'll make the changes accordingly.